sentry-sample/src/androidTest/java/io/sentry/sample/MainActivityTest.kt

```diff
@@ -0,0 +1,95 @@
+package io.sentry.sample
+
+import android.util.Log
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.activityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import io.sentry.core.protocol.SentryId
+import java.util.Timer
+import java.util.concurrent.CountDownLatch
+import kotlin.concurrent.schedule
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+import kotlin.test.assertTrue
+import okhttp3.ResponseBody
+import org.junit.Rule
+import org.junit.runner.RunWith
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+@LargeTest
+@RunWith(AndroidJUnit4::class)
+class MainActivityTest {
+
+    @get:Rule
+    val rule = activityScenarioRule<MainActivity>()
+
+    @Test
+    fun mainActivityTest() {
+        onView(withId(R.id.send_message)).perform(click())
+        onView(withId(R.id.capture_exception)).perform(click())
+        onView(withId(R.id.breadcrumb)).perform(click())
+        onView(withId(R.id.native_capture)).perform(click())
+
+//        val retrofit = Retrofit.Builder()
+//            .baseUrl("https://sentry.io/")
+//            .client(OkHttpClient.Builder().addInterceptor { chain ->
+//                // token from https://sentry.io/settings/account/api/auth-tokens/
+//                val request = chain.request().newBuilder().addHeader("Authorization", "Bearer ${TOKEN}}").build()
```

Comment:
@bruno-garcia do you know if its possible to generate a token on demand? cus we can't read from a system env as this runs on the emulator.
also, it's not "nice" to leak the token on the code, Reverse engineering is too easy at this point.

Reply:
How would we get the generated token?
I agree it's not ideal to embed the key in the apk. We should be able to provide an apk alone and test it out since the testing code is not part of the apk